ServerCore/Helpers/MailHelper.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Mail;
+using Microsoft.Extensions.Configuration;
+
+/// <summary>
+/// Helper class for sending mail with Mailjet.
+/// </summary>
+public class MailHelper
+{
+    private string PublicSecret;
+    private string PrivateSecret;
+
+    public bool Enabled { get; private set; }
+    private bool IsDev;
+
+    private const string FromAddress = "msph@puzzlehunt.org";
+    private const string ToAddressForBCC = "msph-bcc@puzzlehunt.org";
+    
+    public static MailHelper Singleton { get; private set; }
+
+    public static void initialize(IConfiguration configuration, bool isDev)
+    {
+        Singleton = new MailHelper(configuration, isDev);
+    }
+
+    private MailHelper(IConfiguration configuration, bool isDev)
+	{
+        try
+        {
+            PublicSecret = configuration["Mailjet-Public"];
+            PrivateSecret = configuration["Mailjet-Private"];
+            Enabled = true;
+            IsDev = isDev;
+        }
+        catch
+        {
+            Enabled = false;
+        }
+    }
+
+    public void Send(string recipient, string subject, string body)
+    {
+        Send(new List<string> { recipient }, false, subject, body);
+    }
+
+    public void Send(IEnumerable<string> recipients, bool bcc, string subject, string body)
+    {
+        if (!Enabled)
+        {
+            if (!IsDev)
+            {
+                throw new InvalidOperationException("Mail was not configured.");
+            }
+            WriteMailToConsole(recipients, subject, body);
+            return;
+        }
+        MailMessage msg = new MailMessage();
+
+        msg.From = new MailAddress(FromAddress);
+
+        if (bcc)
+        {
+            msg.To.Add(new MailAddress(ToAddressForBCC));
+            foreach (string recipient in recipients) {
+                msg.Bcc.Add(new MailAddress(recipient));
+            }
+        }
+        else
+        {
+            foreach (string recipient in recipients) {
+                msg.To.Add(new MailAddress(recipient));
+            }
+        }
+
+        msg.Subject = subject;
+        msg.Body = body;
+
+        SmtpClient client = new SmtpClient("in.mailjet.com", 587);
```

Comment:
Is this specific choice of port something we need to document? No worries if not. I know there are multiple SMTP ports but I do not know enough to know the differences between them.

Reply:
That's the port that Mailjet tells us to use. I will add a link to their documentation.

Reply:
Actually, I have no idea how to fix. I will cherry-pick the commit over to a new branch and send out a new review after addressing the issues.